R/query.R

```diff
@@ -34,33 +34,18 @@
 #'obama_pageviews <- article_pageviews(article = "Barack_Obama")
 #'
 #'@export
-article_pageviews <- function(project = "en.wikipedia", article = "R (programming language)",
-                              platform = "all", user_type = "all",
-                              start = "2015100100", end = NULL, reformat = TRUE, ...){
+article_pageviews <- function(project = "en.wikipedia", article = "R (programming language)"
+                              , platform = "all", user_type = "all", granularity = "daily"
+                              , start = "2015100100", end = NULL, reformat = TRUE, ...){
 
-
-  article <- gsub(x = article, pattern = " ", replacement = "_", fixed = TRUE)
-  if(length(article) > 1){
-    return(lapply(article, article_pageviews, project = project, platform = platform,
-                  user_type = user_type, start = start, end = end, ...))
+  if(granularity != "daily"){
+    warning("Granularities besides daily are not currently supported by the `article_pageviews` function.")
   }
 
-  # Handle timestamps
-  start <- pageview_timestamps(start)
-  if(is.null(end)){
-    end <- start
-  } else {
-    end <- pageview_timestamps(end)
-  }
-
-  # Construct parameters
-  parameters <- paste("per-article", project, ifelse(platform == "all", "all-access", platform),
-                      ifelse(user_type == "all", "all-agents", user_type), curl::curl_escape(article),
-                      "daily", start, end, sep = "/")
-
-  # Run and return
-  data <- pv_query(parameters, reformat, ...)
+  article <- gsub(x = article, pattern = " ", replacement = "_", fixed = TRUE)
```

Comment:
escaping is about more than just spaces. This isn't an adequate replacement for full encoding.